src/shared/tests/pipes.test.tsx

```diff
@@ -0,0 +1,247 @@
+    expect(listPipe(exampleWorkPackage3.dependencies, wbsPipe)).toBe('1.12.0, 2.5.0');
+    expect(listPipe(exampleAllWbsNums, wbsPipe)).toBe(
+      '1.1.1, 2.7.3, 1.12.0, 2.5.0, 1.1.1, 1.7.1, 1.7.3'
+    );
+  });
+
+  test('Formatting Rules', () => {
+    expect(listPipe(exampleProject1.rules, (str: string) => str)).toBe('EV3.5.2');
+    expect(listPipe(exampleProject2.rules, (str: string) => str)).toBe('T12.3.2, T8.2.6');
+    expect(listPipe(exampleProject3.rules, (str: string) => str)).toBe(
+      'EV3.5.2, EV1.4.7, EV6.3.10'
+    );
+    expect(listPipe(exampleProject5.rules, (str: string) => str)).toBe(
+      'EV3.5.2, T12.3.2, T8.2.6, EV1.4.7, EV6.3.10'
+    );
+  });
+
+  test('Formatting Users', () => {
+    expect(listPipe(exampleAllUsers, fullNamePipe)).toBe(
+      'Thomas Emrax, Joe Shmoe, Joe Blow, Amy Smith, Rachel Barmatha, Emily Bendara, Jackson James'
+    );
+  });
+
+  test('For Other Cases', () => {
+    expect(listPipe([1, 2, 3, 4], (num: number) => `${num}`)).toBe('1, 2, 3, 4');
+    expect(listPipe([], (num: number) => `${num}`)).toBe('');
+    expect(listPipe([1], (num: number) => `${num}`)).toBe('1');
+  });
+});
+
+describe('Formatting End Date Tests', () => {
+  test('with dummy data', () => {
+    expect(endDatePipe(exampleWorkPackage1.startDate, exampleWorkPackage1.duration)).toBe(
+      '1/22/2021'
+    );
+    expect(endDatePipe(exampleWorkPackage2.startDate, exampleWorkPackage2.duration)).toBe(
+      '2/26/2021'
+    );
+    expect(endDatePipe(exampleWorkPackage3.startDate, exampleWorkPackage3.duration)).toBe(
+      '1/15/2021'
+    );
+  });
+
+  test('with edge dates', () => {
+    expect(endDatePipe(new Date('12/25/20'), 3)).toBe('1/15/2021');
+    expect(endDatePipe(new Date('1/3/21'), 3)).toBe('1/24/2021');
+    expect(endDatePipe(new Date('3/1/21'), 10)).toBe('5/10/2021');
+  });
+});
+
+describe('Formatting Links Tests', () => {
+  test('with known websites', () => {
+    expect(linkPipe('google', 'www.google.com')).toStrictEqual(
+      <a href={'www.google.com'}>{'google'}</a>
+    );
+    expect(linkPipe('facebook', 'www.facebook.com')).toStrictEqual(
+      <a href={'www.facebook.com'}>{'facebook'}</a>
+    );
+    expect(linkPipe('github', 'www.github.com')).toStrictEqual(
+      <a href={'www.github.com'}>{'github'}</a>
+    );
+  });
+});
+
+describe('Formatting Weeks Tests', () => {
+  test('with 1', () => {
+    expect(weeksPipe(1)).toBe('1 week');
+  });
+  test('with int > 1', () => {
+    expect(weeksPipe(2)).toBe('2 weeks');
+  });
+  test('with 0', () => {
+    expect(weeksPipe(0)).toBe('0 weeks');
+  });
+  test('with decimal', () => {
+    expect(weeksPipe(1.5)).toBe('1.5 weeks');
+  });
+  test('with hexadecimal', () => {
+    expect(weeksPipe(0x37cf)).toBe('14287 weeks');
+  });
+  test('with octal', () => {
+    expect(weeksPipe(0o377)).toBe('255 weeks');
+  });
+  test('with binary', () => {
+    expect(weeksPipe(0b111001)).toBe('57 weeks');
```

Comment:
I love this, and it's kinda funny cuz it's valid, but this is the kinda test case that would probably be a little much on co-op. 

You can leave it in here, but typically you want to bias towards the minimal number of tests which cover a couple of each base cases and one for each major edge case and no more. This is because more tests = more work to maintain them / edit them when the application changes.